src/sentry/rules/conditions/event_frequency.py

```diff
@@ -125,3 +127,73 @@ def query_hook(self, event, start, end, environment_id):
             environment_id=environment_id,
             use_cache=True,
         )[event.group_id]
+
+
+percent_intervals = {
+    "1m": ("one minute", timedelta(minutes=1)),
+    "5m": ("five minutes", timedelta(minutes=5)),
+    "10m": ("ten minutes", timedelta(minutes=10)),
+    "30m": ("30 minutes", timedelta(minutes=30)),
+    "1h": ("one hour", timedelta(minutes=60)),
+}
+
+
+class EventFrequencyPercentForm(EventFrequencyForm):
+    intervals = percent_intervals
+    interval = forms.ChoiceField(
+        choices=[
+            (key, label)
+            for key, (label, duration) in sorted(
+                percent_intervals.items(),
+                key=lambda key____label__duration: key____label__duration[1][1],
+            )
+        ]
+    )
+    value = forms.IntegerField(widget=forms.TextInput())
+
+
+class EventFrequencyPercentCondition(BaseEventFrequencyCondition):
+    label = "The issue has more errors than {value} percent of sessions in {interval}"
+
+    def __init__(self, *args, **kwargs):
+        self.intervals = percent_intervals
+        self.form_cls = EventFrequencyPercentForm
+        super().__init__(*args, **kwargs)
+
+    def query_hook(self, event, start, end, environment_id):
+        cache_key = f"r.c.spc:{environment_id}-{start.replace(minute=0, second=0, microsecond=0, tzinfo=None)}-{end.replace(minute=0, second=0, microsecond=0, tzinfo=None)}".replace(
```

Comment:
Now that I look at this cache key, I'm not actually sure if it's right. Since start/end can be for different time ranges (1 minute, 5 minutes, etc), aren't the different filters going to end up stomping on each other because we drop off the minute part here?

My initial thought here was that we'd be fetching this for the last hour and caching that, and using the same value across all filters. If we want to handle it this way, then we need to make sure we vary the cache by filter type

Reply:
Aaah you're completely right. Your idea gets around that problem well too. I thought the delay was built in to the snuba side but I think I was wrong about that. Caching on interval here works better I think and resolves the edge case of when start/end are on the boundary of an hour. Gonna do that.

Reply:
Implemented!